### ISSUE TITLE
Sample codes to incorporate an AmazonDynamoDB in a .net core web application

### ISSUE THREAD

**Issue by kawaihk** [aws/aws-sdk-net#934]
I made a sample .net core web application using Visual Studio 2017.
And I created an AmazonDynamoDB and put a table with a few rows of data according to the Developer Guide.
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GettingStarted.NET.html

I would like to incorporate the AmazonDynamoDB in the web application.
I read the article about "Configuring the AWS SDK for .NET with .NET Core"
I understood the dependency injection. 
https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-netcore.html

But I could not find an article with sample codes of a web application that adapted an AmazonDynamoDB.
For example, how to create Razor Pages displaying the data using a table in .NET Core
Could you recommend sample codes in C# ?

Regards

**Comment by kawaihk** [comment]
I found the article with sample codes of a web application that adapted an AmazonDynamoDB After I posted this issue. I created a simple application that uses the Entity Framework and Amazon DynamoDB LocalDB to store and display data using Visual Studio 2017 According to the article.
[HOWTO Build an ASP.NET CORE Web App on AWS Running Linux with DynamoDB – Part 1](http://blogs.catapultsystems.com/khayes/archive/2016/12/09/howto-build-an-asp-net-core-web-app-on-aws-running-linux-with-dynamodb-part-1/)

When I accessed http://localhost:port/Books , A security error occurred.

> System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.Http.WinHttpException: A security error occurred
>   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>   at System.Threading.Tasks.RendezvousAwaitable`1.GetResult()
>   at System.Net.Http.WinHttpHandler.<StartRequest>d__105.MoveNext()

I found the issue about the same error. The article said SDK is failing to resolve the credentials. 
[AWSSDK.Extensions.NETCore.Setup -"Unable to find credentials" happens  #828](https://github.com/aws/aws-sdk-net/issues/828)

I don't think I put a credentials file (~/.aws/credentials) where SDK expected. I understand putting my credentials file in my Windows-User's home directory with aws configure. But I don't know a Windows-User Who executes the web application.
Where(What Windows-User) will I put a credentials file?

**Comment by kawaihk** [comment]
I found another article with sample codes of a web a web application that adapted an AmazonDynamoDB.  And Thanks to his advice, I switched the Local DynamoDB Table to the real DynamoDB Table table on AWS. My web app worked!!
[net-core-api-dynamodb-dependency-injection](http://underscorehao.net/2018/03/net-core-api-dynamodb-dependency-injection/)

There are my codes of the web app on GitHub.   I solved the issue. So I close the issue!
[my sample web app ](https://github.com/kawaihk/an-.NET-CORE-Web-App-With-Amazon-DynamoDB)
